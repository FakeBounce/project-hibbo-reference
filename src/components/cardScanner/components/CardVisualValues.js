import React from 'react';
import { string } from 'prop-types';

import { getTranslations } from 'utils/i18n';

import { StyledContainerBasic } from 'styledComponents/containers';

import CardNumberPreview from './CardNumberPreview';

import * as Styles from '../styles';

const CardVisualValues = ({
  name,
  cardNumber,
  expiracyMonth,
  expiracyYear,
  crypto,
}) => {
  return (
    <Styles.SContainer>
      <Styles.SCardWrapper>
        <Styles.SLabel>
          {getTranslations('cardScanner.visual.cardNumber')}
        </Styles.SLabel>

        <CardNumberPreview cardNumber={cardNumber} />
      </Styles.SCardWrapper>

      <Styles.SDateWrapper>
        <StyledContainerBasic>
          <Styles.SLabel>
            {getTranslations('cardScanner.visual.date')}/{getTranslations('cardScanner.visual.month')}
          </Styles.SLabel>
          <Styles.SText>
            {expiracyMonth !== '' ? expiracyMonth : 'XX'}
            /
            {expiracyYear !== '' ? expiracyYear : 'XX'}
          </Styles.SText>
        </StyledContainerBasic>

        <StyledContainerBasic>
          <Styles.SLabel>
            {getTranslations('cardScanner.visual.cvv')}
          </Styles.SLabel>
          <Styles.SText>{crypto !== '' ? crypto : 'XXX'}</Styles.SText>
        </StyledContainerBasic>
      </Styles.SDateWrapper>

      <Styles.SNameWrapper>
        <Styles.SLabel>
          {getTranslations('cardScanner.visual.cardHolder')}
        </Styles.SLabel>
        <Styles.SText>{name !== '' ? name : 'XXXXXX'}</Styles.SText>
      </Styles.SNameWrapper>
    </Styles.SContainer>
  );
};

CardVisualValues.propTypes = {
  name: string.isRequired,
  cardNumber: string.isRequired,
  expiracyMonth: string.isRequired,
  expiracyYear: string.isRequired,
  crypto: string.isRequired,
};

export default CardVisualValues;
