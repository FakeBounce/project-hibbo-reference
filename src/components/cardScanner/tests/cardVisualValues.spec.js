import React from 'react';
import renderer from 'react-test-renderer';

import CardVisualValues from '../components/CardVisualValues';

describe('CardVisualValues', () => {
  let component = renderer.create(
    <CardVisualValues
      cardNumber=""
      expiracyMonth=""
      expiracyYear=""
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render the component for null cvv', () => {
    component = renderer.create(
      <CardVisualValues
        cardNumber=""
        expiracyMonth={0}
        expiracyYear={0}
      />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
