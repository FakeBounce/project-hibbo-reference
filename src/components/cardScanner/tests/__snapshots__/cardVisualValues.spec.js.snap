// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardVisualValues should render the component 1`] = `
<View
  style={undefined}
>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
  <View
    style={undefined}
  >
    <Image
      source={1}
      style={undefined}
    />
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        EXPIRE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        AT END
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
      /
      
    </Text>
  </View>
</View>
`;

exports[`CardVisualValues should render the component for null cvv 1`] = `
<View
  style={undefined}
>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
  <View
    style={undefined}
  >
    <Image
      source={1}
      style={undefined}
    />
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        EXPIRE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        AT END
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      -
      /
      -
    </Text>
  </View>
</View>
`;
